### PR TITLE
Enable jBallerina finite type test

### LINKED ISSUE #24111: Enable jBallerina finite type test
**Description:**
$subject

**Suggested Labels (optional):**
-Area/Language

**Suggested Assignees (optional):**
-@rdulmina

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$subject

Fixes #24111

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -3655,8 +3662,13 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign) {
         } else if (type == SyntaxKind.DECIMAL_FLOATING_POINT_LITERAL) {
             //TODO: Check effect of mapping negative(-) numbers as unary-expr
             typeTag = NumericLiteralSupport.isDecimalDiscriminated(textValue) ? TypeTags.DECIMAL : TypeTags.FLOAT;
-            value = textValue;
-            originalValue = textValue;
+            if (isFiniteType) {
+                value = textValue.replaceAll("[f,d,/+]", "");
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:3666]: no need to do replaceAll, which may be slower, also why are there `,` commas between chars.
`. replaceFirst("[fd]$", "");`

Comment (manuranga) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:3666]: Oh is this replacing the leading `+` as well? and what about `/` I though escaping is done by `\\`

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/finite/FiniteTypeTest.java [hunk 2]
```diff
@@ -362,11 +362,11 @@ public void testFiniteTypeWithDiscriminatedMembers() {
...
-    @Test(description = "Test finite type where typename is espaed identifier", groups = { "brokenOnSpecDeviation" })
-    public void testEscapedTypeName() {
-        BValue[] returns = BRunUtil.invoke(result, "testEscapedTypeName");
-        Assert.assertEquals(returns[0].stringValue(), "-");
-    }
+//    @Test(description = "Test finite type where typename is espaed identifier", groups = { "brokenOnSpecDeviation" })
+//    public void testEscapedTypeName() {
+//        BValue[] returns = BRunUtil.invoke(result, "testEscapedTypeName");
+//        Assert.assertEquals(returns[0].stringValue(), "-");
+//    }
```
Comment (SupunS) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/finite/FiniteTypeTest.java:369]: Any reason for commenting out?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 3]
```diff
@@ -630,7 +630,14 @@ public BLangNode transform(UnionTypeDescriptorNode unionTypeDescriptorNode) {
...
         for (TypeDescriptorNode finiteTypeEl : finiteTypeElements) {
             SingletonTypeDescriptorNode singletonTypeNode = (SingletonTypeDescriptorNode) finiteTypeEl;
-            BLangLiteral literal = createSimpleLiteral(singletonTypeNode.simpleContExprNode());
+            BLangLiteral literal;
+            Node simpleContExprNode = singletonTypeNode.simpleContExprNode();
+            if (simpleContExprNode.kind() == SyntaxKind.UNARY_EXPRESSION) {
+                UnaryExpressionNode unaryExpr = (UnaryExpressionNode) simpleContExprNode;
+                literal = createSimpleLiteral(unaryExpr.expression(), unaryExpr.unaryOperator().kind(), true);
+            } else {
+                literal = createSimpleLiteral(simpleContExprNode, SyntaxKind.NONE, true);
```
Comment (SupunS) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:639]: I think this is already handles inside `createSimpleLiteral()` method?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 1 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: removed comma and /. :-)

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/finite/FiniteTypeTest.java (hunk 2 above)
Comment (rdulmina) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/finite/FiniteTypeTest.java]: Uncommented

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 3 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: removed

**Pull request merged**